infoschema/slow_log.go

```diff
@@ -83,12 +84,25 @@ var slowQueryCols = []columnInfo{
 }
 
 func dataForSlowLog(ctx sessionctx.Context) ([][]types.Datum, error) {
-	return parseSlowLogFile(ctx.GetSessionVars().Location(), ctx.GetSessionVars().SlowQueryFile)
+	var hasProcessPriv bool
+	if pm := privilege.GetPrivilegeManager(ctx); pm != nil {
+		if pm.RequestVerification(ctx.GetSessionVars().ActiveRoles, "", "", "", mysql.ProcessPriv) {
+			hasProcessPriv = true
+		}
+	}
+	user := ctx.GetSessionVars().User
+	checkValid := func(userName string) bool {
+		if !hasProcessPriv && user != nil && userName != user.Username {
```

Comment:
```suggestion
		if !hasProcessPriv || user == nil || userName != user.Username ?
```

Reply:
false returned when ?
1: !hasProcessPriv
2: user == nil
3: userName != user.Username

Reply:
No, this condition means:
* If the user has `ProcessPriv ` privilege, then return true.
* otherwise, only return true when `user` not nil and `user.Username` equal to the  taget `userName`.